### ISSUE TITLE
 Web - IOU – Unable to create IOU with non-existing user via 1:1 conversation

### ISSUE THREAD

**Issue by izarutskaya** [Expensify/App#33151]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 1.4.13-0
**Reproducible in staging?:** Y
**Reproducible in production?:** N
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Expensify/Expensify Issue URL:**
**Issue reported by:** Applause-Internal Team
**Slack conversation:** @

## Action Performed:
1. Open staging.new.expensify.com
2. Navigate to FAB> Start Chat
3. Enter email address of non-existing user
4. Click on "+" button from created chat
5. Select Request money> Add amount and click Next

## Expected Result:
Request button should be active and Money request should be created successfully with non-existing user just like creating IOU with non-existing user from FAB button

## Actual Result:
Unable to create IOU with non-existing user via 1:1 conversation. Request button is inactive

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Native
- [ ] Android: mWeb Chrome
- [ ] iOS: Native
- [ ] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

https://github.com/Expensify/App/assets/115492554/1dd811f1-d3e2-442e-9682-4020ac37cb90

</details>

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by Julesssss** [comment]
Likely caused by fixes for the big refactor PR...

**Comment by paultsimura** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.


### What is the root cause of that problem?
We are filtering out the hidden participants (who have no login but have an `accountID`).

### What changes do you think we should make in order to solve the problem?
```js
.filter((participant) => !!participant.login || !!participant.accountID || !!participant.text)
```

https://github.com/Expensify/App/blob/fd71747303cd8df7fc1dfc9b010752e5635b2684/src/pages/iou/request/step/IOURequestStepConfirmation.js#L88

### What alternative solutions did you explore? (Optional)

**Comment by Julesssss** [comment]
Replacing C+ as this is urgent

**Comment by allroundexperts** [comment]
@paultsimura's [proposal](https://github.com/Expensify/App/issues/33151#issuecomment-1857742657) looks good to me. Simple fix.

🎀 👀 🎀 C+ reviewed

**Comment by Julesssss** [comment]
Thank you.

**Comment by paultsimura** [comment]
On it

**Comment by situchan** [comment]
What's the offending PR?

**Comment by paultsimura** [comment]
> What's the offending PR?

The big money request flow refactoring: https://github.com/Expensify/App/pull/28618

**Comment by situchan** [comment]
> > What's the offending PR?
> 
> The big money request flow refactoring: #28618

That's not the answer I expected. This is deploy blocker and #28618 was deployed to production already

**Comment by paultsimura** [comment]
> That's not the answer I expected

But that's the only one I have. Also, this is not a deploy blocker – it's happening in prod as well

**Comment by allroundexperts** [comment]
@paultsimura Can you open a PR even if its draft?

**Comment by situchan** [comment]
@paultsimura's solution is wrong.

Here's correct fix:
```diff
             .map((participant) => { 
                 const isPolicyExpenseChat = lodashGet(participant, 'isPolicyExpenseChat', false); 
                 return isPolicyExpenseChat ? OptionsListUtils.getPolicyExpenseReportOption(participant) : OptionsListUtils.getParticipantsOption(participant, personalDetails); 
             }) 
-            .filter((participant) => !!participant.login || !!participant.text) 
             .value(), 
```
So the refactor reverted #32166 which removed this line

Here's old logic:
https://github.com/Expensify/App/blob/173582ff506202ca32671cc0a7acea1f440dddb0/src/pages/iou/steps/MoneyRequestConfirmPage.js#L71-L78

**Comment by Julesssss** [comment]
CP in progress

**Comment by Julesssss** [comment]
CP'd and working as expected

**Comment by joekaufmanexpensify** [comment]
Per [this](https://expensify.slack.com/archives/C01GTK53T8Q/p1703082403934039), payment may be delayed a few days here because of the holidays. @allroundexperts if you have time over the next week to handle the BZ checklist too, that'd be great!

**Comment by joekaufmanexpensify** [comment]
@allroundexperts could you please complete the checklist so we can issue payment?

**Comment by joekaufmanexpensify** [comment]
[Bumped in Slack](https://expensify.slack.com/archives/C02NK2DQWUX/p1704208634755849)

**Comment by joekaufmanexpensify** [comment]
Still discussing in Slack

**Comment by paultsimura** [comment]
Does the BZ usually affect the Contributor payout as well?

**Comment by joekaufmanexpensify** [comment]
Yep! As soon as the BZ checklist is complete, we'll issue payment to everyone.

**Comment by allroundexperts** [comment]
## Checklist
1. https://github.com/Expensify/App/pull/28618
2. https://github.com/Expensify/App/pull/28618/files#r1440647938
3. N/A
4. A regression test would be helpful.

## Regression test
1. Open the app and navigate to FAB -> Start Chat
2. Enter email address of non-existing user
3. Click on "+" button from created chat
4. Select Request money -> Add amount and click Next
5. Verify that request button should be active and money request should be created successfully with non-existing user just like creating IOU with non-existing user from FAB button.

Do we 👍 or 👎 ?

**Comment by joekaufmanexpensify** [comment]
We need to make the following payments: 

- @paultsimura - $500 for PR. Paid via Upwork. 
- @allroundexperts - $500 for C+ review. Paid via NewDot.

**Comment by joekaufmanexpensify** [comment]
@paultsimura $500 sent and contract ended!

**Comment by joekaufmanexpensify** [comment]
@allroundexperts could you please request $500 and confirm here once you've done that?

**Comment by JmillsExpensify** [comment]
$500 payment to @allroundexperts based on [this comment](https://github.com/Expensify/App/issues/33151#issuecomment-1875964844).
